### ISSUE TITLE
RAG and Tool usage , at the same time

### ISSUE THREAD

**Issue by aditya11ad** [langflow-ai/langflow#1955]
i have a telecom use case:

1. RAG to general broadband related queries. 
eg.:  
**user >>** "What is the maximum speed your broadband offers?"
**bot  <<** "We offer 1gbps"

2. tool usage (function calling) for : 
_eg_.:  
**user >>** "show me my mobile data balance?"
<bot calls a function **get_mobile_data_balance(mobile_number)**>
**bot <<** "plz enter  ur moile number"
**user >>** 9876678978
<call send_msg(9876678978)>
**bot<<** "plz enter OTP sent to u"
**user >>** 8822
<call otp_validation_and_return_data(8822)>
**bot<<** "ur data balance is 23GB"
---
so basically it is a mixture of both , and the agent decides what to invoke :
my way of doing it:
`tools = [retrive_tool, get_mobile_data_balance]`

**Q1**. How can i achieve this in langflow?

**Q2**. After invoking get_mobile_data_balance, how can i maintain the control flow with proper conversation as shown in _eg_.

**Comment by yamonkjd** [comment]
you can run an agent in the Agents section, but you will need to define the two tools you mentioned using custom components.

Contrary to DosuBot's response, if you declare the output type of the custom component as AgentExecutor, you can declare and control the agent directly as you wish.

**Comment by carlosrcoelho** [comment]
Hi @aditya11ad,

Just wanted to check in and see if you still need any help with anything.

**Comment by carlosrcoelho** [comment]
Thank you for your contribution! This issue will be closed. If you have any questions or encounter another problem, please open a new issue and we will be ready to assist you.
